### DISCUSSION TITLE
validation loss on eval_every_steps ?

### DISCUSSION THREAD

**Discussion by leo90v** [nyu-mll/jiant#1341] [Q&A]
Hi,

I'm using the eval_every_steps parameter for early stopping. I noticed it also stocks the metrics on the train_val.zlog file but would it be possible to also stock the loss at that evaluation like it is done for the best model on the val_metrics.json file ? 

I would like to compare my training and evaluation losses if possible.

Thanks !

**Comment by sleepinyourhat** [comment]
I'm not sure— @zphang, ming taking a look?

**Comment by zphang** [comment]
Hi,

This function call is where the the metrics are extracted for each of the val_result dictionaries. 

https://github.com/nyu-mll/jiant/blob/235f646bd292e2f5bde444379f487d1d50f035a4/jiant/proj/main/metarunner.py#L231-L233

However, the dictionary should also contain a key for the loss value. So you can try modifying code are that segment to save the loss as well.
